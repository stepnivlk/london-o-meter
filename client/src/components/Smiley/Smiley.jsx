// @flow

import React from 'react';

type Props = {
  isPositive: boolean,
}

const negativePath = () =>
  <path d="M0,0 A40,0 10 0,0 65,0" fill='none' stroke='black' strokeWidth={5} />

const positivePath = () =>
  <path d="M0,0 A40,40 10 0,0 65,0" fill='none' stroke='black' strokeWidth={5} />

const Smiley = ({ isPositive }: Props) => (
  <g transform='translate(-50,-50)'>
    <circle r={5} fill='black' id='c' cx={25} cy={30} />
    <circle r={5} fill='black' id='c' cx={75} cy={30} />
    <g transform='translate(15,65)'>
      {
        isPositive ? positivePath() : negativePath()
      }
    </g>
  </g>
);

export default Smiley;
