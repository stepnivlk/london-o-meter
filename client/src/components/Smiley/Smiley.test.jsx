import React from 'react';
import renderer from 'react-test-renderer';
import Smiley from './Smiley';

it('matches a snapshot', () => {
  const component = renderer.create(
    <Smiley />
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
