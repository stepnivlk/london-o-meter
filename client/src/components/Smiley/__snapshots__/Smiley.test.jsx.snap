// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<g
  transform="translate(-50,-50)"
>
  <circle
    cx={25}
    cy={30}
    fill="black"
    id="c"
    r={5}
  />
  <circle
    cx={75}
    cy={30}
    fill="black"
    id="c"
    r={5}
  />
  <g
    transform="translate(15,65)"
  >
    <path
      d="M0,0 A40,0 10 0,0 65,0"
      fill="none"
      stroke="black"
      strokeWidth={5}
    />
  </g>
</g>
`;
